apis/v1alpha2/validation/httproute.go

```diff
@@ -47,10 +47,17 @@ func ValidateHTTPRoute(route *gatewayv1a2.HTTPRoute) field.ErrorList {
 // HTTPRoute specification.
 func validateHTTPRouteSpec(spec *gatewayv1a2.HTTPRouteSpec, path *field.Path) field.ErrorList {
 	var errs field.ErrorList
+	for i, rule := range spec.Rules {
+		errs = append(errs, validateHTTPRouteUniqueFilters(rule.Filters, path.Child("rules").Index(i))...)
+		errs = append(errs, validateHTTPRouteFilterTypeMatchesValue(rule.Filters, path.Child("rules").Index(i))...)
```

Comment:
This is already a great simplification, but can you take it one step further and just have a generic `validateHTTPRouteFilters` func instead that covers both of these? Would save us from looping through the same list of filters 2x.

Reply:
I kept them separate to have a clear boundary for each validation but I can see the inefficiency.  I'll simplify 👍🏾